presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -458,16 +500,108 @@ else if (isCovariantParameterPosition(commonSuperTypeBase.get(), i)) {
         return Optional.of(new TypeSignature(commonSuperTypeBase.get(), typeParameters.build()));
     }
 
-    /*
-     * Given signature with unmatched parameters return filtered one.
+    /**
+     * In case of coercion between un-parametrized types (DOUBLE, BIGINT, ...)
+     * we only need to know whether coercion is allowed (true|false).
+     * But in case of coercion to the type with long parameters, parameters of the result type must be calculated.
+     * Such coercion rules must be implemented within this method.
+     * <p>
+     * e.g.:
+     * LONG -> DECIMAL = DECIMAL(19,0)
```

Comment:
This is not syntax suggar. `DECIMAL(x,y)` is still valid for the function signatures. In order to bind some parameters with coercion to such signatures we must handle unspecified `DECIMAL(x,y)` in `getCommonSupperType` and `canCoerce` method. So the `canCoerce(BIGINT, DECIMAL(x,y))` returns `TRUE`.